server/controllers/moleculesImporter.js

```diff
@@ -0,0 +1,262 @@
+import Logger, { addErrorTitle } from "../global/Logger.js";
+import { analyzeData } from "../global/molecules_analyzer/moleculesAnalyzer.js";
+import { createSqlToInsertAllData } from "../global/molecules_importer/moleculesImporter.js";
+import { parseMoleculesFromCsv } from "../global/molecules_parser/Parser.js";
+
+/**
+ *
+ * @api {post} /import/molecules Import a csv file of molecules data
+ * @apiName ImportMolecules
+ * @apiGroup Import
+ * 
+ * @apiPermission LoggedIn
+ * @apiPermission Admin
+ * @apiDescription Import a csv file to update the molecules data, the format of the request must be multipart/form-data ! 
+ * 
+ * @apiParam  {File} file The csv file
+ * @apiParam {string} confirmed If "true", the data will be imported, otherwise they will be only tested
+ * 
+ * @apiSuccess (200) {string} message Message explaining what was done
+ * @apiSuccess (200) {object[]} warnings Array of warnings
+ * @apiSuccess (200) {object} warnings.warning a warning
+ * @apiSuccess (200) {number} warnings.warning.code The warning code
+ * @apiSuccess (200) {string} warnings.warning.message The warning message
+ * @apiSuccess (200) {boolean} imported Boolean telling if data are imported
+ *
+ * @apiSuccessExample Success-Response:
+ *  {
+      "message": "File tested but not imported",
+      "warnings": [
+        {
+          "type": 3,
+          "message": "Ces valeurs de \"classes\" sont très proches : \"PENICILLINES A\", \"PENICILLINES\""
+        },
+        {
+          "type": 3,
+          "message": "Ces valeurs de \"indications\" sont très proches : \"VIH\", \"VHB\""
+        }
+      ],
+      "imported": false
+    }
+  
+ * @apiError (400) BadFormattedFile The csv file is bad formatted
+ * @apiErrorExample BadFormattedFile Error-Response:
+    {
+      "message": {
+        "message": "Bad formatted file",
+        "errors": [
+          {
+            "code": 1,
+            "message": "Colonne manquante : 'FORMULE_CHIMIQUE'"
+          },
+          {
+            "code": 4,
+            "message": "Colonne invalide : 'INDICATIO' (col. 10)"
+          },
+          {
+            "code": 4,
+            "message": "Colonne invalide : 'FORMULE CHIMIQUE' (col. 19)"
+          }
+        ],
+        "imported": false
+      }
+    }   
+ * @apiError (400) BadFileType The file is not csv
+ * @apiError (400) MissingFile No file provided
+ * @apiUse ErrorServer
+ *
+ */
+function importMolecules(req, _res) {
+  const res = new HttpResponseWrapper(_res);
+
+  const {
+    _uploadedFileName: filename,
+    _uploadedFileExtension: extension,
+    _uploadedFileDirectory: directory,
+    confirmed,
+  } = req.body;
+
+  if (!filename) {
+    return res.sendUsageError(400, "Missing file");
+  }
+
+  const deleteUploadedFile = () =>
+    deleteFile(path.resolve(directory, filename)).catch(Logger.error);
+
+  if (extension !== "csv") {
+    res.sendUsageError(400, "Invalid file format : must be csv ");
```

Comment:
This error message should be in French
```suggestion
    res.sendUsageError(400, "Invalid file format : must be csv ");
```